src/Blog/Domain/OpenGraph/ImageGenerator.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Blog\Domain\OpenGraph;
+
+use App\Blog\Domain\Model\Article;
+
+/**
+ * Generate an OpenGraph image for a given Article.
+ */
+interface ImageGenerator
+{
+    public const int WIDTH = 1200;
+
+    public const int HEIGHT = 630;
+
+    public const int SPACING = 24;
+
+    public const int AUTHOR_AVATAR_SIZE = 64;
+
+    public const int TITLE_FONT_SIZE = 80;
+
+    public const int AUTHOR_FONT_SIZE = 34;
+
+    public const int LOGO_HEIGHT = 32;
+
+    /**
+     * @return string The image content in JPEG format.
```

Comment:
First I went for WEBP format, but for a better comptability with OpenGraph reader tools and websites etc, I believe that's better to use a more common format like jpeg.